projects/conductor/tests/004-searchEvents.test.ts

```diff
@@ -0,0 +1,183 @@
+import { test } from '../utils/playwright'
+import testData from '../utils/testData'
+import { expect } from '@playwright/test'
+import { delay } from '../utils/delay'
+
+test.describe('SearchEvents suite', () => {
+  test('with valid api key and limit', async ({ assert, identify }) => {
+    const { visitorId } = await identify({
+      auth: testData.credentials.maxFeaturesUS,
+    })
+
+    await assert.thatResponsesMatch('searchEvents', {
+      visitorId,
+      limit: 10,
+      apiKey: testData.credentials.maxFeaturesUS.privateKey,
+      region: testData.credentials.maxFeaturesUS.region,
+    })
+  })
+
+  test('with invalid limit', async ({ assert }) => {
+    await assert.thatResponseMatch({
+      expectedStatusCode: 400,
+      expectedResponse: {
+        error: {
+          code: 'RequestCannotBeParsed',
+          message: 'invalid limit',
+        },
+      },
+      callback: (api) =>
+        api.searchEvents({
+          apiKey: testData.credentials.maxFeaturesUS.privateKey,
+          region: testData.credentials.maxFeaturesUS.region,
+          limit: 0,
+        }),
+    })
+  })
+
+  test('with partial params', async ({ identify, assert }) => {
+    const { visitorId } = await identify({
+      auth: testData.credentials.maxFeaturesUS,
+    })
+
+    await assert.thatResponseMatch({
+      expectedStatusCode: 200,
+      callback: (api) =>
+        api.searchEvents({
+          apiKey: testData.credentials.maxFeaturesUS.privateKey,
+          region: testData.credentials.maxFeaturesUS.region,
+          limit: 10,
+          bot: 'good',
+          visitorId,
+        }),
+    })
+  })
+
+  test('with all params', async ({ identify, assert, fingerprintApi }) => {
+    const { visitorId, requestId } = await identify({
+      auth: testData.credentials.maxFeaturesUS,
+      linkedId: 'test',
+    })
+
+    const { data: event } = await fingerprintApi.getEvent({
+      apiKey: testData.credentials.maxFeaturesUS.privateKey,
+      region: testData.credentials.maxFeaturesUS.region,
+      requestId,
+    })
+
+    const start = new Date()
+    start.setHours(start.getHours() - 1)
+
+    const end = new Date()
+    end.setHours(end.getHours() + 1)
+
+    await assert.thatResponseMatch({
+      expectedStatusCode: 200,
+      callback: (api) =>
+        api.searchEvents({
+          apiKey: testData.credentials.maxFeaturesUS.privateKey,
+          region: testData.credentials.maxFeaturesUS.region,
+          limit: 10,
+          bot: 'good',
+          visitorId,
+          linkedId: 'test',
+          start: start.getTime(),
+          end: end.getTime(),
+          ipAddress: `${event.products.identification.data.ip}/24`,
+          reverse: false,
```

Comment:
We also need test case with passing `true`value.

Reply:
Done!